middleware/metrics.go

```diff
@@ -0,0 +1,71 @@
+package middleware
+
+import (
+	"net/http"
+	"strconv"
+
+	"github.com/gorilla/mux"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+type responseWriter struct {
+	http.ResponseWriter
+	statusCode int
+}
+
+func NewResponseWriter(w http.ResponseWriter) *responseWriter {
+	return &responseWriter{w, http.StatusOK}
+}
+
+func (rw *responseWriter) WriteHeader(code int) {
+	rw.statusCode = code
+	rw.ResponseWriter.WriteHeader(code)
+}
+
+var totalRequests = prometheus.NewCounterVec(
```

Comment:
I haven't used Prometheus in go, but is it standard to make so many top level vars?

Reply:
Looking around it seems like it might be. If so I wonder if it would be cleaner to put the metric declarations in their own file for each package? The registration can also be done in the init() I think. 

Open to other thoughts here. Just imagining what this looks like when we have many places with metrics.

Reply:
I was gonna say "as far as I can tell, yes," haha. I had taken them out of the init because I wasn't sure we wanted the registers to be called just from importing the middleware. No real preference either way.

Cleanliness wise, it's hard to say. Having the metrics definitions in their own file makes them less obtrusive, but also adds a layer of abstraction to understanding the code. Same as above, no real preference. I guess if pushed, I generally default towards things that are used together living together, but that might get real messy.